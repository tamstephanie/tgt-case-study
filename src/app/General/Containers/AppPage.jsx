import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import {Typography, withStyles} from "@material-ui/core";

class AppPage extends PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="app-page-content">
                <Typography variant="h5"><b>{this.props.title}</b></Typography>
                {this.props.children}
            </div>
        );
    }
}

AppPage.propTypes = {
    /**
     * Text to display at the top of the page
     * @type {String}
     */
    title: PropTypes.string.isRequired,
};

export default AppPage;