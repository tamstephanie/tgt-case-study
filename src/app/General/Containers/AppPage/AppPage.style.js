export const AppPageStyle = (theme) => ({
    header: {
        backgroundColor: theme.palette.secondary.main,
        marginBottom: "50px",
        minHeight: "250px",
        "& .app-title-card": {
            padding: "125px 0px 25px 65px"
        }
    },
    appTitle: {
        backgroundColor: "rgb(255, 255, 255, 0.7)",
        borderRadius: "0px",
        boxShadow: "none",
        maxWidth: "fit-content",
        "& .MuiCardContent-root": {
            color: theme.palette.secondary.main,
            padding: "16px",
            textAlign: "center"
        }
    }
});