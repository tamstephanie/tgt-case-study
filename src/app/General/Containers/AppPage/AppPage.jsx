import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import {Card, CardContent, Container, Typography, withStyles} from "@material-ui/core";

import {AppPageStyle} from "./AppPage.style";

class AppPage extends PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        let {classes} = this.props;
        return (
            <div className="app-page">
                <Container className={classes.header} maxWidth="xl">
                    <div className="app-title-card">
                        <Card className={classes.appTitle} raised={false}>
                            <CardContent>
                                <Typography variant="h1"><b>{this.props.title}</b></Typography>
                            </CardContent>
                        </Card>
                    </div>
                </Container>
                <Container className={classes.content}>
                    <Typography className="subtitle" variant="h2" gutterBottom>
                        <b>{this.props.subtitle}</b>
                    </Typography>
                    {this.props.children}
                </Container>
            </div>
        );
    }
}

AppPage.propTypes = {
    /**
     * Text to display at the top of the page
     * @type {String}
     */
    title: PropTypes.string.isRequired,
    /**
     * JSON object of CSS styling, which is then applied to the corresponding component(s)
     * @type {JSS}
     * @optional
     */
    classes: PropTypes.object,
    /**
     * Text to display right below the header at the top of the page's content
     * @type {String}
     * @optional
     */
    subtitle: PropTypes.string,
};

export default withStyles(AppPageStyle)(AppPage);