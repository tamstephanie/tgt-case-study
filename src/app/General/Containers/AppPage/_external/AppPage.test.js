import React from "react";
import chai from "chai";
import {mount} from "enzyme";
import {create} from "react-test-renderer";

import AppPage from "../AppPage";

describe("AppPage", () => {
    it("should match the rendered snapshot", () => {
        let page = create(<AppPage title="Sample Page" />);
        expect(page).toMatchSnapshot();
    });

    it("should always display a page title", () => {
        let page = mount(<AppPage title="Sample Page" />);

        // Expect a page title to exist and match the provided text
        chai.expect(page.find(".MuiTypography-h1")).to.have.lengthOf(1);
        chai.expect(page.find(".MuiTypography-h1").text()).to.equal("Sample Page");

        // Expect there to be no subtitle since none was given
        chai.expect(page.find(".subtitle")).to.have.lengthOf(0);
    });

    it("should display a subtitle, if provided", () => {
        let page = mount(<AppPage title="Sample Page" subtitle="Sample Subtitle" />);
        
        // Expect a subtitle to exist and match the provided text
        chai.expect(page.find(".MuiTypography-h2")).to.have.lengthOf(1);
        chai.expect(page.find(".MuiTypography-h2").text()).to.equal("Sample Subtitle");
    });
});