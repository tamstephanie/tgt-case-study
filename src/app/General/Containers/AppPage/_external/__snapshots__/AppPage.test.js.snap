// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPage should match the rendered snapshot 1`] = `
<div
  className="app-page"
>
  <div
    className="MuiContainer-root AppPage-header-1 header MuiContainer-maxWidthXl"
  >
    <div
      className="app-title-card"
    >
      <div
        className="MuiPaper-root MuiCard-root AppPage-appTitle-2 app-title MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          <h1
            className="MuiTypography-root MuiTypography-h1"
          >
            <b>
              Sample Page
            </b>
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiContainer-root AppPage-content-3 content MuiContainer-maxWidthLg"
  />
</div>
`;
